Utils/FaviconFetcher.swift

```diff
@@ -128,8 +128,16 @@ public class FaviconFetcher : NSObject, NSXMLParserDelegate {
                                 iconType = nil
                         }
                     }
+
+                    guard let href = link.attribute("href") where iconType != nil else {
+                        return
```

Comment:
`return`? maybe I'm reading this wrong, but wouldn't `continue` be better so we can keep iterating the links we've found?